backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -137,23 +78,41 @@ private MimeMessage createEmail(String subject, String bodyText) throws Messagin
 
     email.setFrom(new InternetAddress(APPLICATION_EMAIL));
     email.addRecipient(RecipientType.TO, new InternetAddress(recipientEmail));
-    email.setSubject(subject);
+    email.setSubject(subjectText);
     email.setText(bodyText);
     return email;
   }
 
   /** Function that access its api and using it sends an email */
-  //   TODO(#35): Create a dummy email for ad lib itself to send emails.
   //   TODO(#36): Replace body to send real link to user instead of generic.
-  public void notifyUser() throws MessagingException, IOException, GeneralSecurityException {
-    NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+  public void sendMatchEmail(String firstMatchRecipientName, String firstMatchRecipientEmail)
+      throws MessagingException, IOException {
+    MimeMessage email =
+        createEmail(
+            firstMatchRecipientEmail,
+            "Ad-lib: We found you a match !",
+            String.format(
+                " We found you a match with %s "
+                    + " Check your calendar for your meeting event,"
+                    + " and feel free to join the Meet call now!\n",
+                firstMatchRecipientName));
+    Message messageWithEmail = createMessageWithEmail(email);
+    service.users().messages().send("me", messageWithEmail).execute();
+  }
+
+  /** Function that access its api and using it sends an email */
+  public void sendExpiredEmail(String expiredRecipientUsername, String expiredRecipientEmail)
+      throws MessagingException, IOException {
     MimeMessage email =
         createEmail(
-            "Ad-Lib Meeting Found",
-            " Hey "
-                + recipientName
-                + " Please Join your Ad-Lib meeting via the link below : \n"
-                + " http://meet.google.com/new");
+            expiredRecipientEmail,
+            "Ad-lib: Sorry, we couldn't find you a match!",
+            String.format(
+                " Sorry %s We couldn't find a match for you this time, but we encourage you to "
```

Comment:
What's the %s?

Reply:
It is just a regular expression to express that there will be a string, the string is the second argument in the String.format function which is the recipient username

Reply:
Absolutely right, though it's a string placeholder and not a regular expression.